src/python/pants/engine/internals/build_files.py

```diff
@@ -387,16 +389,91 @@ def _extract_env_vars(
     )
 
     # Process synthetic targets.
-    for address_map in address_maps:
-        for synthetic in synthetic_address_maps:
-            synthetic.process_declared_targets(address_map)
-            synthetic.apply_defaults(frozen_defaults)
+
+    def apply_defaults(tgt: TargetAdaptor) -> TargetAdaptor:
+        default_values = frozen_defaults.get(tgt.type_alias)
+        if default_values is None:
+            return tgt
+        return tgt.with_new_kwargs(**default_values, **tgt.kwargs)
```

Comment:
this will fail with a `TypeError` if there is a overridden default value:
```
In [6]: f(**arg, **blarg)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 1
----> 1 f(**arg, **blarg)

TypeError: __main__.f() got multiple values for keyword argument 'foo'

```

Reply:
Good catch! Fixed.